chromium/background-scripts/background.js

```diff
@@ -187,37 +187,60 @@ chrome.browserAction.onClicked.addListener(e => {
 });
 
 
-
 /**
- * Add a listener for removed tabs
+ * A centralized storage for browsing data within the browser session.
  */
-function AppliedRulesets() {
-  this.active_tab_rules = new Map();
-  this.active_tab_main_frames = new Map();
+function BrowserSession() {
+  this.tabs = new Map();
+  this.requests = new Map();
 
   let that = this;
```

Comment:
Is there any reason for this? Wouldn't just turning the next function into an arrow function make `this` a part of its closure?